### DISCUSSION TITLE
Only select the code block content when pressing Cmd + A inside the node

### DISCUSSION THREAD

**Discussion by bcbogdan** [outline/outline#6495] [Ideas]
Hi everyone,

I'm sorry in advance if this was already discussed and I missed a previous issue.

## Problem

I've been writing a couple of documents that make use of **Mermaid code blocks**.
And for me the usual flow is:
1. Go to mermaid.live and get some sample code
2. Define your first diagram
3. When it's time to define another mermaid block:
   - create the empty block 
   - click on the previous one and press `Cmd+A` (this is where the flow breaks)
   - then copy and paste it in the newly created one

This is probably a pattern that I got from using Notion, but the _separated_ UI of the mermaid block also helps to make you feel that only the code will be copied.

## Solution

As the title says, when your **cursor** is inside a **code block** and you press `Mod-A` the editor would only select the text that's inside the block. If you press the keyboard shortcut again the selection will include the entire document.
It's basically how Notion works.

I already have a working example on my local build so if you're ok with the proposal I can go ahead and submit a PR.

## Offtopic
Congrats on the entire product! I've had a great experience using it for the last week. 

Thank you,
Bogdan

**Comment by tommoor** [comment]
Easy fix, will address this week
